app/models/scheduled_event.rb

```diff
@@ -1,12 +1,62 @@
 class ScheduledEvent
   include Mongoid::Document
+  include Mongoid::Timestamps
 
   field :type, type: String
   field :event_name, type: String
-  field :start_date, type: Date
-  field :one_time, type: Boolean, default: false
-  field :recurring_rules, type: String
-  field :offset_rule, type: Integer, default: none
+  field :start_time, type: Date
+  field :one_time, type: Boolean, default: true
+  field :recurring_rules, type: Hash
+  field :offset_rule, type: Integer, default: 0
 
-  validates_presence_of :type, :event_name, :one_time, :start_date
+  embeds_many :event_exceptions
+
+  validates_presence_of :type, :event_name, :one_time, :start_time, :message => "fields type, event_name can't be empty"
+
+  EVENT_TYPES = %W(system holiday)
+  HOLIDAYS = %W(new_year martinluthor_birthdday washingtons_day memorial_day independence_day
+                Labour_day columbus_day veterans_day Christmas Thanksgiving_day)
```

Comment:
labour_day => labor_day (defer to american spelling, since it's an american holiday)
Christmas => christmas
Thanskgiving_day => thanksgiving
martinluthor_birthdday => martin_luther_king_birthday
washingtons_day => presidents_day